common/lib/xmodule/xmodule/lti_module.py

```diff
@@ -841,6 +855,17 @@ def get_client_key_secret(self):
                 return key, secret
         return '', ''
 
+    def is_past_due(self):
+        """
+        Is it now past this problem's due date, including grace period?
+        """
+        due_date = self.due  # pylint: disable=no-member
+        if self.graceperiod is not None and due_date:  # pylint: disable=no-member
```

Comment:
The fields are in https://github.com/edx/edx-platform/blob/master/common/lib/xmodule/xmodule/modulestore/inheritance.py -- to implement inheritance, they get mixed into blocks in the course hierarchy.